### PR TITLE
Change: Warn when the opensearch instance size is set small

### PR DESCRIPTION
**Pull request by marcvberg**
#### Description of changes

Print out a warning of the OpenSearch instance size is t2.small

#### Issue #, if available

#8123

#### Description of how you validated changes
Ran amplify-dev locally

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/extensions/amplify-helpers/searchable-push-checks.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { EnvironmentDoesNotExistError, exitOnNextTick, stateManager, $TSAny, $TSContext } from 'amplify-cli-core';
+import { ResourceConstants } from 'graphql-transformer-common';
+import { printer } from 'amplify-prompts';
+import { readJsonFile } from './read-json-file';
+
+
+export function searchablePushChecks(context: $TSContext): void {
+  const currEnv = context.amplify.getEnvInfo().envName;
+  const teamProviderInfoFilePath = context.amplify.pathManager.getProviderInfoFilePath();
+  const teamProviderInfo = readJsonFile(teamProviderInfoFilePath);
```
Comment (lazpavel) [inline packages/amplify-cli/src/extensions/amplify-helpers/searchable-push-checks.ts:10]: nit: can we use stateManager.getTeamProviderInfo

File: packages/amplify-cli/src/extensions/amplify-helpers/searchable-push-checks.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  const apiCategory = teamProviderInfo[currEnv]?.categories?.api;
+  const instanceType = apiCategory ? apiCategory[ResourceConstants.PARAMETERS.ElasticsearchInstanceType] : null;
+  if (!instanceType || instanceType === 't2.small.elasticsearch') {
```
Comment (lazpavel) [inline packages/amplify-cli/src/extensions/amplify-helpers/searchable-push-checks.ts:13]: shouldn't this warning appear only if the project is using api with @searchable?

Comment (marcvberg) [reply packages/amplify-cli/src/extensions/amplify-helpers/searchable-push-checks.ts:13]: Fixed

File: packages/amplify-cli/src/extensions/amplify-helpers/searchable-push-checks.ts (hunk 1 above)
Comment (marcvberg) [reply packages/amplify-cli/src/extensions/amplify-helpers/searchable-push-checks.ts:10]: Fixed

**Review by lazpavel — APPROVED** [review]
LGTM

**Pull request merged**